### PR TITLE
add support for precision==64 in Trainer

### LINKED ISSUE #2497: Add Float64 support
Float64 precision is currently not supported. 
Why? This should be pretty straightforward, raising an Exception if the device or other configurations are not compatible with it.
Are you planning to add support soon? 
What is the best workaround currently? Will it work if I do it manually via `model.to(torch.float64)` and similar for each dataloader or are there some caveats?

### PR DESCRIPTION
**Pull request by richardk53**
This PR fixes #2497 

cast model to float64 / float32, depending on precision.
In the collection of tensors in a batch, cast each float tensor also to float64 / float32.

### CODE REVIEW

**Comment by Borda** [comment]
@richardk53 how is it going, mind align with master?

**Comment by Borda** [comment]
@richardk53 mind rebase on master and resolve conflicts?

File: pytorch_lightning/utilities/casting.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+
+def cast_model_to_precision(model):
+    if model.precision in [16, 32]:
```
[the remark this answers is not in the document]
Comment (justusschock) [reply pytorch_lightning/utilities/casting.py:19]: yeah, but mixed precision is different then half precision. Just adding a comment should be fine here

**Comment by edenlightning** [comment]
@richardk53 friendly ping :)

**Comment by richardk53** [comment]
sorry, I have been pretty busy. 
I will have time to update this to the master in 2 weeks if there is still interest for this feature...

**Comment by Borda** [comment]
@richardk53 how is it going here, we have moved this feature to next 1.2, can we get it done by that time?

**Comment by Borda** [comment]
@richardk53 how is it going, shall we push this further?

**Comment by richardk53** [comment]
Sorry, let's better close this issue, I cannot work on this.

**Comment by tchaton** [comment]
Any updates on this PR ?

**Pull request closed without merging**